### PR TITLE
Add derive credential interface

### PR DESCRIPTION
**Pull request by sudeshrshetty**
added derive credential feature based on #81

### CODE REVIEW

File: docs/index.html [hunk 1]
```diff
@@ -1461,7 +1461,33 @@ <h3>Prove</h3>
...
+          Takes a Verifiable Credentials id, a JSON-LD <a href="https://w3c-ccg.github.io/ldp-bbs2020/#dfn-frame">frame</a> for selective disclosure and an
+          <code>options</code> object.
+        </p>
+
+        <p>Produces a Derived Verifiable Credential.</p>
+
+        <p>
+          Here are the options that can be used to derive a verifiable credential:
+        <ul>
+          <li><a href="https://w3c-ccg.github.io/ldp-bbs2020/#nonce">nonce </a> </li>
```
Comment (TallTed) [inline docs/index.html:1480]:
```suggestion
          <li><a href="https://w3c-ccg.github.io/ldp-bbs2020/#nonce">nonce</a></li>
```

**Review by TallTed — CHANGES_REQUESTED** [review]

**Comment by OR13** [comment]
@sudeshrshetty I will look to you to review @TallTed but this looks good to me. thanks!

**Review by OR13 — APPROVED** [review]

**Comment by OR13** [comment]
conflicts after recent merges

File: docs/index.html (hunk 1 above)
Comment (sudeshrshetty) [reply docs/index.html:1480]: fixed

File: docs/index.html [hunk 2]
```diff
@@ -1461,7 +1461,33 @@ <h3>Prove</h3>
...
-            let verifiableCredential = wallet.Prove(ids, options)
+            let verifiableCredential = wallet.prove(ids, options)
+        </pre>
+      </section>
+
+      <section>
+        <h3>Derive</h3>
+        <p>
+          Takes a Verifiable Credentials id, a JSON-LD <a href="https://w3c-ccg.github.io/ldp-bbs2020/#dfn-frame">frame</a> for selective disclosure and an
+          <code>options</code> object.
```
[the remark this answers is not in the document]
Comment (sudeshrshetty) [reply docs/index.html:1472]: fixed

File: docs/index.html [hunk 3]
```diff
@@ -1461,7 +1461,33 @@ <h3>Prove</h3>
...
+        </ul>
+        </p>
+
+        <pre class="example highlight" title="Derive credential example">
+            let id = "http://example.edu/credentials/1872" // id of the verifiable credential
+            let frame = {...} // a JSON-LD frame
+            let options = {
+              nonce: "0b4e419a-1410-4739-a58d-b37f4db10181"
+             }
+            let derivedCredential = wallet.derive(id, frame, options)
```
[the remark this answers is not in the document]
Comment (sudeshrshetty) [reply docs/index.html:1490]: fixed

**Comment by sudeshrshetty** [comment]
> conflicts after recent merges

@OR13  rebased

**Review by OR13 — APPROVED** [review]

**Comment by sudeshrshetty** [comment]
@TallTed Can you please have a look at resolved comments?

**Comment by TallTed** [comment]
@sudeshrshetty - They're fine.

**Review by kimdhamilton — APPROVED** [review]

**Comment by sudeshrshetty** [comment]
@OR13 Can we merge this one?

**Pull request merged**